### PR TITLE
Added Budget.java and two subclasses

### PR DESCRIPTION
**Pull request by hmartyscu**

### CODE REVIEW

File: src/main/java/entities/Budget.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
+package entities;
```
Comment (WaseeA) [inline src/main/java/entities/Budget.java:1]: Please add a class description

File: src/main/java/entities/GroceryBudget.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package entities;
```
Comment (WaseeA) [inline src/main/java/entities/GroceryBudget.java:1]: Another class description may be helpful, but I don't believe it's necessary.

File: src/main/java/entities/SubscriptionBudget.java [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package entities;
```
Comment (WaseeA) [inline src/main/java/entities/SubscriptionBudget.java:1]: Another class description may be helpful, but I don't believe it's necessary.

**Review by WaseeA — COMMENTED** [review]
Most comments are just about documentation.

File: src/main/java/entities/Budget.java [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+
+    /**
+     * Change the quantity of the item with the given name and of the given category to the given quantity.
+     *
+     * @param category    the category of the item
+     * @param item        the name of the item
+     * @param newQuantity the new quantity of the item
+     * @return whether the quantity was changed
+     */
+    public boolean changeQuantity(String category, String item, int newQuantity) {
```
[the remark this answers is not in the document]
Comment (hmartyscu) [reply src/main/java/entities/Budget.java:117]: Perhaps add a quantities attribute to Budget, private final Map<String, Map<String, Integer>> quantities, to be updated when an item is added/removed along with budget

**Pull request closed without merging**